### PR TITLE
ceph: 9.2.0 -> 12.2.2

### PR DESCRIPTION
**Pull request by adevress**
- Update Ceph to last version
- Solve security vulnerability of the old version
- Provide shared version of snappy, required for Ceph

###### Motivation for this change
- Make Ceph usable again
- Provide shared version for snappy

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by andir** [comment]
Thank you for working on this. You must also update the sha256 of ceph otherwise you will build the old version with a new name ;)

File: pkgs/development/libraries/snappy/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, cmake }:
+{ stdenv, fetchFromGitHub, cmake, sharedLibrary ? false }:
```
Comment (Mic92) [inline pkgs/development/libraries/snappy/default.nix]: I would rather build shared libraries by default. Which is what we prefer in nixpkgs.

Comment (Mic92) [reply pkgs/development/libraries/snappy/default.nix]: This variable does not have an effect on this derivation, was this added by mistake?

**Comment by adevress** [comment]
Hi @Mic92,

Ok this patch comes from an old nixpkgs branch we have in our side (BBP).

I just realized it diverged way more I expected, and that the recommended way to build Ceph is cmake nowadays.

This would imply way more modifications, let me have a look to that and I come back to you.

**Comment by adevress** [comment]
ping @vcunat  @Mic92 

Ok guys, problems fixed.


- The build reciep for Ceph has been entirely switched to CMake. It greatly simplified the build process.
- External dependencies to boost and RocksDB are enforced.
- This pull request requires #34662 and #34661 to be merged
- Ceph is now built with RocksDB as default backend
- Ceph now has its own python environment, avoiding ugly hack to get its python tool / doc generation working.
- Every output is separated correctly

**Comment by srhb** [comment]
@lejonet Might be interesting to test this as the base for your ceph module :)h

**Comment by lejonet** [comment]
@srhb Yeah, I will do that, seems like @adevress did a thorough work through of the build, I only fixed what I needed to get it to compile at all, to develop the ceph module.

Will fork and apply my module ontop and try it out

**Comment by lejonet** [comment]
@adevress I've tried your branch on a nixos system, with sandbox enabled and all binaries compile and run fine
I just had to make a tiny patch for the ceph source code and your package that you can see here: https://github.com/lejonet/nixpkgs/commit/a782a50ad160bc184e5ba972a1eb8cf77746bcce

**Comment by joachifm** [comment]
@GrahamcOfBorg build ceph bareos

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**